container/impl-base/src/test/java/org/jboss/arquillian/container/impl/client/container/DeploymentExceptionHandlerTestCase.java

```diff
@@ -97,6 +97,21 @@ public void shouldCallDeploymentTransformers() throws Exception {
         Mockito.verify(transformer, Mockito.times(1)).transform(Mockito.isA(Exception.class));
     }
 
+    @Test
+    public void shouldCallDeploymentTransformersWithEmptyCauseException() throws Exception {
+        TestExceptionDeployThrower.shouldThrow =
+            new DeploymentException("Could not handle ba", null);
+        Mockito.when(serviceLoader.all(DeploymentExceptionTransformer.class))
+            .thenReturn(Collections.singletonList(transformer));
+
+        fire(new DeployDeployment(
+            container,
+            new Deployment(new DeploymentDescription("test", ShrinkWrap.create(JavaArchive.class))
+                .setExpectedException(DeploymentException.class))));
+
+        Mockito.verify(transformer, Mockito.times(1)).transform(Mockito.isA(Exception.class));
```

Comment:
I know that you want to verify that null is not passed to the `transform` method, but I believe that we can make this check more strict. We could also check that the exception passed to the method is the same as was thrown during the deployment: 
`Mockito.verify(transformer, Mockito.times(1)).transform(TestExceptionDeployThrower.shouldThrow);`
or at least that the type of the exception is `DeploymentException`:
`Mockito.verify(transformer, Mockito.times(1)).transform(Mockito.isA(DeploymentException.class));`

Reply:
> ... we can make this check more strict

Fair enough, will do.

> or at least that the type of the exception is DeploymentException:...

That should be verified by the line above - `.setExpectedException(DeploymentException.class)`
If you get any other exception, it blows up.